test/end-to-end/stock/stock.aggregate_consumption.page.js

```diff
@@ -0,0 +1,164 @@
+/* global element, by */
+const FU = require('../shared/FormUtils');
+const GU = require('../shared/GridUtils');
+const components = require('../shared/components');
+const SharedStockPage = require('./stock.shared.page');
+
+function StockAggregateConsumptionPage() {
+  const page = this;
+  const gridId = 'aggregated-consumption-grid';
+  const modalGridId = 'lots-grid';
+
+  // change current depot
+  page.changeDepot = SharedStockPage.setDepot;
+
+  // the grid id
+  page.gridId = gridId;
+  page.modalGridId = modalGridId;
+
+  // create modal
+  page.showCreateModal = async isNewRequisition => {
+    const selector = isNewRequisition ? '[data-method="create-record"]' : '[data-method="create-other-record"]';
+    await $(selector).click();
+  };
+
+  page.showSearchModal = () => {
+    return $('[data-method="search"]').click();
+  };
+
+  page.setFiscalPeriod = async (fiscalYear, period) => {
+    await components.fiscalYearPeriodSelect.set(fiscalYear, period);
+  };
+
+  // depot select
+  page.setDepot = depot => {
+    return components.depotSelect.set(depot, 'depot-supplier');
+  };
+
+  page.setRows = async number => {
```

Comment:
Let's call this `addRows` instead of `setRows`, since we are clicking the "add rows" button on the page.